tests/main/views/test_reset_password.py

```diff
@@ -435,7 +435,7 @@ def test_token_created_before_last_updated_password_cannot_be_used(self):
 
         assert res.status_code == 200
         document = html.fromstring(res.get_data(as_text=True))
-        error_selector = cssselect.CSSSelector('div.banner-destructive-without-action')
+        error_selector = cssselect.CSSSelector('div.dm-alert.dm-alert--error')
```

Comment:
Can these assertion lines be handled with `self.assert_flashes(..., 'error')` as per the other tests?

Reply:
Not without rewriting the test quite a bit.

`assert_flashes()` checks for the presence of the flash message in the session data. The flash message is popped from the session data when it is shown, which I think is why the other tests don't follow the redirect, so the page showing the flash message is never reached.

Since this test uses `follow_redirects=True` when POSTing the data, the flash message is shown, and `assert_flashes()` raises an error `nothing flashed`. I'm not sure why this test is different to the others, but looking at the test name `test_token_created_before_last_updated_password_cannot_be_used` I think the flash message test might be entirely incidental to the actual thing being tested...

Reply:
Ahh it's one of those. Fair enough.